### PR TITLE
fix: pagination issue with several React hooks

### PR DESCRIPTION
**Pull request by cesarenaldi**
What to review in this PR:
- ApolloClient upgrade
- `config.ts` changes
- one non-paginated hook: `usePublication`
- one lazy non-paginated hook: `useLazyProfile`
- one paginated hook: `useProfileFollowers`
- one lazy list hook: `useLazyPublications`
- adds missing `useCurrencies` paginated field policy
- some of the "odd ones": `useInvitedProfiles`, `useRevenueFromFollow` (just `$fxRateFor` var used)
- removed unused queries in `modules.graphql`
- removed code from previous variable injection approach (start from `createTypePolicies.ts`)

### CODE REVIEW

File: packages/api-bindings/package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "typescript": "5.2.2"
   },
   "peerDependencies": {
-    "@apollo/client": "~3.8.7",
+    "@apollo/client": "^3.8.10",
```
Comment (krzysu) [inline packages/api-bindings/package.json:77]: is this version number correct?

File: packages/react/src/shared.tsx [hunk 2]
```diff
@@ -79,7 +72,7 @@ export function createSharedDependencies(userConfig: BaseConfig): SharedDependen
 
   // apollo client
   const apolloClient = createLensApolloClient({
-    queryParams: config.params,
+    connectToDevTools: config.debug,
```
Comment (krzysu) [inline packages/react/src/shared.tsx:75]: nice!

**Review by krzysu — APPROVED** [review]
reviewed, tested, LGTM 👍

File: packages/api-bindings/package.json (hunk 1 above)
Comment (cesarenaldi) [reply packages/api-bindings/package.json]: Not affecting the final result, but prefer to fix it. Good catch.

**Pull request merged**